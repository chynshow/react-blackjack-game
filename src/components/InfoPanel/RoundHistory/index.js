import React, { useState } from 'react';

const RoundHistory = ({ roundHistory }) => {
  const [showPanel, setShowPanel] = useState(true);

  return (
    <>
      {roundHistory.length > 0 && (
        <div
          className={`${
            showPanel
              ? 'c-info-panel__item c-round-history c-round-history--active'
              : 'c-info-panel__item c-round-history'
          }`}
        >
          <h4 className='c-round-history__title'>Round History</h4>
          {roundHistory.map((item, idx) => (
            <div className='c-round-history__item' key={idx}>
              <div className='c-round-history__numberOfRound'>
                Round: {item.round}
              </div>
              <div className='c-round-history__credit'>
                Credit: {item.credit}$
              </div>
              <div className='c-round-history__bet'>Bet: {item.bet}$</div>
              <div className='c-round-history__cards'>
                <span className='c-round-history__dealer-cards'>
                  Dealer: {item.score?.dealerScore}
                </span>
                {item.cards?.dealerCards.map((card) => (
                  <img
                    className='c-round-history__img'
                    src={card.image}
                    key={card.code}
                    alt={card.code}
                  />
                ))}

                <span className='c-round-history__player-cards'>
                  Player: {item.score?.playerScore}
                </span>
                {item.cards?.playerCards.map((card) => (
                  <img
                    className='c-round-history__img'
                    src={card.image}
                    key={card.code}
                    alt={card.code}
                  />
                ))}
              </div>
            </div>
          ))}
          <button
            className='c-btn c-round-history__show-history-btn'
            onClick={() => setShowPanel(!showPanel)}
          >
            Hide
          </button>
        </div>
      )}
    </>
  );
};

export default React.memo(RoundHistory);
