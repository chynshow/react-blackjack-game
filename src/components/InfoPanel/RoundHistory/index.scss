.c-round-history {
  border-radius: 5px;
  color: $color-dark;
  background: $color-light;
  animation: scale 1000ms linear both;

  &__title {
    padding: 0.5rem;
    text-align: center;
    font-weight: normal;
    font-size: 1.5rem;
  }

  &__item {
    margin: 0.5rem;
    animation: scale 1000ms linear both;
  }

  &__cards {
    display: flex;
    padding: 0.5rem 0;
  }

  &__cards-container {
    display: flex;

    align-items: center;
  }

  &__dealer-cards {
    margin-right: 0.5rem;
  }
  &__player-cards {
    margin: 0 0.5rem;
  }

  &__img {
    width: 1rem;
    transition: 0.2s ease-in-out;

    &:hover {
      width: 3rem;
    }
  }
}
