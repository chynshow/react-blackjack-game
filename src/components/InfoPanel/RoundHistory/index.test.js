import React from 'react';
import RoundHistory from './index';
import { shallow } from 'enzyme';

it('render RoundHistory component', () => {
  const c = shallow(<RoundHistory />);
  expect(c).toMatchSnapshot();
});
