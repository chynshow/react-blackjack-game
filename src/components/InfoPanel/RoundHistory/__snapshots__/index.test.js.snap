// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render RoundHistory component 1`] = `<Fragment />`;
